src/pages/Search/AdvancedSearchFilters.tsx

```diff
@@ -220,24 +221,38 @@ function AdvancedSearchFilters() {
     const {singleExecution} = useSingleExecution();
     const waitForNavigate = useWaitForNavigation();
     const [reports] = useOnyx(ONYXKEYS.COLLECTION.REPORT);
-
     const [searchAdvancedFilters = {} as SearchAdvancedFiltersForm] = useOnyx(ONYXKEYS.FORMS.SEARCH_ADVANCED_FILTERS_FORM);
     const [cardList = {}] = useOnyx(ONYXKEYS.CARD_LIST);
     const taxRates = getAllTaxRates();
     const personalDetails = usePersonalDetails();
     const currentType = searchAdvancedFilters?.type ?? CONST.SEARCH.DATA_TYPES.EXPENSE;
 
-    const onFormSubmit = () => {
-        const query = SearchUtils.buildQueryStringFromFilterValues(searchAdvancedFilters);
+    const CANNED_SEARCH_QUERY = 'type:expense status:all';
+    const queryString = useMemo(() => SearchUtils.buildQueryStringFromFilterValues(searchAdvancedFilters), [searchAdvancedFilters]);
+
+    const applyFiltersAndNavigate = () => {
         SearchActions.clearAllFilters();
         Navigation.dismissModal();
         Navigation.navigate(
             ROUTES.SEARCH_CENTRAL_PANE.getRoute({
-                query,
+                query: queryString,
             }),
         );
     };
 
+    const onSaveSearch = () => {
+        const queryJSON = SearchUtils.buildSearchQueryJSON(queryString || SearchUtils.buildCannedSearchQuery());
+        if (!queryJSON) {
+            return;
+        }
```

Comment:
I put it since I had this error:

`Type 'SearchQueryJSON | undefined' is not assignable to type 'SearchQueryJSON'.`

That `undefined` value comes from:
`    const queryString = useMemo(() => SearchUtils.buildQueryStringFromFilterValues(searchAdvancedFilters), [searchAdvancedFilters]);
`

I've remove it though and followed another approach.

Reply:
Ah I see. Thanks for clarifying